### PR TITLE
rsyslog: initial addon

### PR DESCRIPTION
**Pull request by awiouy**
@islipfd19 did most of the work ;)

### CODE REVIEW

**Comment by lrusak** [comment]
Can all the libs not be compiled statically?

File: packages/addons/service/rsyslog/package.mk [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+PKG_SHORTDESC="RSYSLOG: a rocket-fast system for log processing."
+PKG_LONGDESC="RSYSLOG ($PKG_VERSION) offers high-performance, great security features and a modular design."
+PKG_AUTORECONF="yes"
+
+PKG_IS_ADDON="yes"
+PKG_ADDON_NAME="rsyslog"
+PKG_ADDON_TYPE="xbmc.service"
+PKG_ADDON_REPOVERSION="8.0"
+
+PKG_CONFIGURE_OPTS_TARGET="--prefix=/storage/.kodi/addons/service.rsyslog \
```
Comment (lrusak) [inline packages/addons/service/rsyslog/package.mk]: This prefix looks wrong

**Comment by awiouy** [comment]
I checked again:
- packages do not build with "--enable-static --disable-shared"
- service does not run without the prefix

**Comment by stefansaraev** [comment]
what's the deal with libbuid, why you cant use one that comes from util-linux ?

**Comment by awiouy** [comment]
Rebase with the following changes:
The external dependencies now build static.
Instructions have been added.
rsyslog expects plugins as shared libraries, --prefix makes sure it finds them outside of /lib.

**Comment by stefansaraev** [comment]
from man 8 rsyslogd
```
ENVIRONMENT
       ...
       RSYSLOG_MODDIR
              Provides the default directory in which loadable modules reside.
```

you should revert prefix and add use Environment=RSYSLOG_MODDIR=/storage/.... in your systemd service

**Comment by awiouy** [comment]
Thank you stefan, i will do that

**Comment by stefansaraev** [comment]
side note: it's most likely that you or your users will not need gcrypt support at all. disabling gcrypt (see configure) may result in "less bumps and rebuilds just because someone bumped gcrypt" in the future. 

when I've done addon packaging in the past, my goal was to only support what "most" users need, and less external (especialy non static) denepdencies that an addon maintainer have no control of. 

but that's just me and my own opinion. sure supporting encrypted network transport has it's use-cases ;)

**Comment by awiouy** [comment]
Is gcrypt not included statically into rsyslogd?
Bumping gcrypt should not affect rsyslogd, or I am missing something?

ldd /storage/.kodi/addons/service.rsyslog/bin/rsyslogd
	linux-vdso.so.1 (0x7ef08000)
	/lib/libarmmem.so (0x76fa3000)
	libz.so.1 => /usr/lib/libz.so.1 (0x76f92000)
	libpthread.so.0 => /lib/libpthread.so.0 (0x76f6a000)
	libdl.so.2 => /lib/libdl.so.2 (0x76f57000)
	libuuid.so.1 => /lib/libuuid.so.1 (0x76fe0000)
	libsystemd.so.0 => /lib/libsystemd.so.0 (0x76ee9000)
	librt.so.1 => /lib/librt.so.1 (0x76ed2000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x76ea6000)
	libc.so.6 => /lib/libc.so.6 (0x76d72000)
	/lib/ld-linux-armhf.so.3 (0x76fb8000)
	libm.so.6 => /lib/libm.so.6 (0x76cf4000)
	libresolv.so.2 => /lib/libresolv.so.2 (0x76cd0000)

**Comment by awiouy** [comment]
I see now:
ldd /storage/.kodi/addons/service.rsyslog/lib/rsyslog/lmcry_gcry.so
ldd: $warning: you do not have execution permission for `/storage/.kodi/addons/service.rsyslog/lib/rsyslog/lmcry_gcry.so'
	linux-vdso.so.1 (0x7eeae000)
	/lib/libarmmem.so (0x76f0d000)
	libgcrypt.so.20 => /usr/lib/libgcrypt.so.20 (0x76e82000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x76e56000)
	libc.so.6 => /lib/libc.so.6 (0x76d22000)
	/lib/ld-linux-armhf.so.3 (0x54b7f000)

**Comment by stefansaraev** [comment]
build.*/toolchain/bin/*-readelf -d is your best friend to check for shlib depends :)

**Comment by awiouy** [comment]
The risk is limited, though
https://github.com/LibreELEC/LibreELEC.tv/commits/ce890b1f04c0932f3d9912eaaf5fab237f6fd38f/packages/security/libgcrypt/package.mk

**Comment by stefansaraev** [comment]
lgtm. this PR makes dagwieers happy again /me runs ;)

File: packages/addons/service/rsyslog/source/bin/rsyslog.init [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with LibreELEC.  If not, see <http://www.gnu.org/licenses/>.
+################################################################################
+
+. /etc/profile
+oe_setup_addon service.rsyslog
+
+mkdir -p "$ADDON_HOME"
```
Comment (stefansaraev) [inline packages/addons/service/rsyslog/source/bin/rsyslog.init]: oe_setup_addon takes care of this

**Comment by islipfd19** [comment]
Please merge. Built and tested for LE 8.0 on WeTek Hub without issues.

**Comment by lrusak** [comment]
Please update the commit dates before we merge this.

**Comment by lrusak** [comment]
Thanks
LGTM

**Pull request merged**